### ISSUE TITLE
Tests fails when using mongoid

### ISSUE THREAD

**Issue by alexmobo** [Netflix/fast_jsonapi#173]
https://github.com/Netflix/fast_jsonapi/blob/18d30f228e09ab4a2b304053d81b183a688e8871/lib/extensions/has_one.rb#L4

When someone uses `mongoid` ORM, the `require 'active_record'` doesn't let tests run crashing with the following error
```
Failure/Error: raise ConnectionNotEstablished, "No connection pool for #{klass}" unless pool

     ActiveRecord::ConnectionNotEstablished:
       No connection pool for ActiveRecord::Base
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:571:in `retrieve_connection'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/connection_handling.rb:113:in `retrieve_connection'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/connection_handling.rb:87:in `connection'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/fixtures.rb:501:in `create_fixtures'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/fixtures.rb:979:in `load_fixtures'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/fixtures.rb:952:in `setup_fixtures'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-4.2.8/lib/active_record/fixtures.rb:826:in `before_setup'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-rails-3.5.2/lib/rspec/rails/adapters.rb:127:in `block (2 levels) in <module:MinitestLifecycleAdapter>'
     # /Users/work/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:443:in `instance_exec'
```

I imagine this error is happening because of the requirement in that file. Don't you think, since that is an extension, the require should be made by the developer's code instead of the gem?

I mean, the `require 'active_record'` is totally unnecessary since active record would be already loaded if the developer is using it, right?

**Comment by alexmobo** [comment]
instead of the requirement
```ruby
if defined?(::ActiveRecord)
  ...
end
```

**Comment by Cakirucho** [comment]
Take a look here #123

**Comment by shishirmk** [comment]
This should be fixed in dev branch now as #123  has been merged. It will go out in the next version.

**Comment by alexmobo** [comment]
Cool! Thanks!
